Tests/SwiftDocCTests/Infrastructure/SnippetResolverTests.swift

```diff
@@ -0,0 +1,272 @@
+            }
+            
+            XCTAssertNil(snippetRenderBlocks.dropFirst(4).first, "There's no more content after the snippets")
+        }
+    }
+    
+    func testWarningsAboutMisspelledSnippetPathsAndMisspelledSlice() async throws {
+        for pathPrefix in optionalPathPrefixes.prefix(1) {
+            let (problems, logOutput, snippetRenderBlocks) = try await makeSnippetContext(
+                snippets: [
+                    makeSnippet(
+                        pathComponents: ["Snippets", "First"],
+                        explanation: """
+                        Some _formatted_ **content** that provides context to the snippet.
+                        """,
+                        code: """
+                        // Some code comment
+                        print("Hello, world!")
+                        """,
+                        slices: [
+                            "comment": 0..<1,
+                            "print":   1..<2,
+                        ]
+                    ),
+                ],
+                rootContent: """
+                @Snippet(path: \(pathPrefix)Frst)
+                
+                @Snippet(path: \(pathPrefix)First, slice: commt)
+                """
+            )
+            
+            // The first snippet has a misspelled path and the second has a misspelled slice
+            XCTAssertEqual(problems.map(\.diagnostic.summary), [
+                "Snippet named 'Frst' couldn't be found",
+                "Slice named 'commt' doesn't exist in snippet 'First'",
+            ])
+            
+            // Verify that the suggested solutions correct the issues.
+            let rootMarkupContent = """
+            # Heading
+            
+            Abstract 
+            
+            ## Subheading 
+            
+            @Snippet(path: \(pathPrefix)Frst)
+            
+            @Snippet(path: \(pathPrefix)First, slice: commt)
+            """
+            do {
+                let snippetPathProblem = try XCTUnwrap(problems.first)
+                let solution = try XCTUnwrap(snippetPathProblem.possibleSolutions.first)
+                let modifiedLines = try solution.applyTo(rootMarkupContent).components(separatedBy: "\n")
+                XCTAssertEqual(modifiedLines[6], "@Snippet(path: \(pathPrefix)First)")
+            }
+            do {
+                let snippetSliceProblem = try XCTUnwrap(problems.last)
+                let solution = try XCTUnwrap(snippetSliceProblem.possibleSolutions.first)
+                let modifiedLines = try solution.applyTo(rootMarkupContent).components(separatedBy: "\n")
+                XCTAssertEqual(modifiedLines[8], "@Snippet(path: \(pathPrefix)First, slice: comment)")
+            }
+            
+            let prefixLength = pathPrefix.count
+            XCTAssertEqual(logOutput, """
+            \u{001B}[1;33mwarning: Snippet named 'Frst' couldn't be found\u{001B}[0;0m
```

Comment:
FYI: I'm not _too_ happy about verifying escape code output like this because it makes the test really hard to read and debug if something goes wrong. Without `highlights` in the diagnostic configuration, there's no visual indication of the line range that the warning applies to (I'm not counting the character range after the file name). I feel that it's important for this test to verify that the correct portions of the line are highlighted for each warning and I didn't feel like it was within score of this PR to update the general diagnostic output. Considering that, I feel like verifying the output with escape codes is better than the alternative for now.